modules/platforms/dotnet/Apache.Ignite.Core/Impl/Binary/BinaryProcessor.cs

```diff
@@ -165,41 +167,39 @@ public BinaryType RegisterEnum(string typeName, IEnumerable<KeyValuePair<string,
         /// Gets the type name by id.
         /// </summary>
         /// <param name="id">The identifier.</param>
+        /// <param name="registerSameJavaType">True if should register type both for dotnet and java platforms.</param>
         /// <returns>Type or null.</returns>
-        public string GetTypeName(int id)
+        public string GetTypeName(int id, bool registerSameJavaType)
         {
-            try
+            return GetTypeName(id, DotNetPlatformId, ex =>
             {
-                return GetTypeName(id, DotNetPlatformId);
-            }
-            catch (BinaryObjectException)
-            {
-                if (!Marshaller.RegisterSameJavaType.Value)
-                    throw;
-            }
+                if (!registerSameJavaType)
+                    throw ex;
```

Comment:
Stack trace gets lost when we rethrow an exception like this. It should be wrapped in another exception, e.g. `new BinaryObjectException("Unknown type ID: " + id, ex)`

Reply:
Fixed